From 76635c1de314e6f83c8e6041ec05d869d32f785e Mon Sep 17 00:00:00 2001
From: Guilhem IMBERTON <guilhem.imberton@intel.com>
Date: Wed, 29 May 2013 19:11:28 +0200
Subject: Build with optimized gcc-4.7 compiler

BZ: 103828

Use gcc 4.7 to build.

Change-Id: I6ba0123235b70e38997fecea267992ae40e637b4
Orig-Change-Id: I146ad9f5ace5343990a96dba019be3551c2c60e5
Category: device enablement
Domain: AOSP
Origin: internal
Upstream-Candidate: no
Signed-off-by: Guilhem IMBERTON <guilhem.imberton@intel.com>
---
 core/combo/TARGET_linux-x86.mk |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/core/combo/TARGET_linux-x86.mk b/core/combo/TARGET_linux-x86.mk
index 18747bd..5bb0571 100644
--- a/core/combo/TARGET_linux-x86.mk
+++ b/core/combo/TARGET_linux-x86.mk
@@ -43,9 +43,15 @@ include $(TARGET_ARCH_SPECIFIC_MAKEFILE)
 
 # You can set TARGET_TOOLS_PREFIX to get gcc from somewhere else
 ifeq ($(strip $(TARGET_TOOLS_PREFIX)),)
+ifeq ($(TARGET_GCC_VERSION),4.7)
+# Use full feature toolchain with x86_64, x32, slm support
+TARGET_TOOLCHAIN_ROOT := prebuilts/gcc/$(HOST_PREBUILT_TAG)/x86/x86_64-linux-android-4.7
+TARGET_TOOLS_PREFIX := $(TARGET_TOOLCHAIN_ROOT)/bin/x86_64-linux-android-
+else
 TARGET_TOOLCHAIN_ROOT := prebuilts/gcc/$(HOST_PREBUILT_TAG)/x86/i686-linux-android-$(TARGET_GCC_VERSION)
 TARGET_TOOLS_PREFIX := $(TARGET_TOOLCHAIN_ROOT)/bin/i686-linux-android-
 endif
+endif
 
 TARGET_CC := $(TARGET_TOOLS_PREFIX)gcc$(HOST_EXECUTABLE_SUFFIX)
 TARGET_CXX := $(TARGET_TOOLS_PREFIX)g++$(HOST_EXECUTABLE_SUFFIX)
-- 
1.7.4.1

